Client/YooChooseNotifier.php

```diff
@@ -236,98 +235,124 @@ public function hideLocation($locationId, $isChild = false)
+        $return = array(
+            'action' => $action,
+            'format' => 'EZ',
+            'uri' => $this->getContentUri($content, $lang),
+            'itemId' => $content->id,
+            'contentTypeId' => $content->contentInfo->contentTypeId,
+        );
 
-        $contentType = $this->repository->sudo(function () use ($contentId, $contentTypeService, $contentService) {
-            $contentType = $this->contentTypeService->loadContentType(
-                $this->contentService
-                    ->loadContent($contentId)
-                    ->contentInfo
-                    ->contentTypeId
-            );
+        if (null !== $lang) {
+            $return['lang'] = $lang;
+        }
+
+        return $return;
+    }
+
+    /**
+     * Checks if content is excluded from supported content types.
+     *
+     * @param \eZ\Publish\API\Repository\Values\Content\Content $content
+     *
+     * @return bool
+     */
+    private function isContentTypeExcluded(Content $content)
+    {
+        $contentType = $this->repository->sudo(function () use ($content) {
+            $contentType = $this->contentTypeService->loadContentType($content->contentInfo->contentTypeId);
 
             return $contentType;
         });
 
-        return $contentType->identifier;
+        return !in_array($contentType->identifier, $this->options['included-content-types']);
     }
 
     /**
-     * Gets ContentType ID based on $contentId.
+     * Gets languageCodes based on $content.
      *
-     * @param mixed $contentId
+     * @param \eZ\Publish\API\Repository\Values\Content\Content $content
+     * @param int|null $versionNo
      *
-     * @return int|null
+     * @return array
      */
-    protected function getContentTypeId($contentId)
+    protected function getLanguageCodes($content, $versionNo = null)
     {
-        $contentTypeId = null;
+        $version = $this->contentService->loadVersionInfo($content->contentInfo, $versionNo);
 
-        try {
-            $contentTypeId = $this->contentService->loadContentInfo($contentId)->contentTypeId;
-        } catch (Exception $e) {
-        }
-
-        return $contentTypeId;
+        return $version->languageCodes;
     }
 
     /**
      * Generates the REST URI of content $contentId.
      *
-     * @param $contentId
+     * @param \eZ\Publish\API\Repository\Values\Content\Content $content
+     * @param $lang
      *
      * @return string
      */
-    protected function getContentUri($contentId)
+    protected function getContentUri(Content $content, $lang = null)
     {
         return sprintf(
-            '%s/api/ezp/v2/content/objects/%s',
+            '%s/api/ezp/v2/ez_recommendation/v1/content/%s%s',
             // @todo normalize in configuration
             $this->options['server-uri'],
-            $contentId
+            $content->id,
+            isset($lang) ? '?lang=' . $lang : ''
+        );
+    }
+
+    /**
+     * Returns the YooChoose notification endpoint.
```

Comment:
you have used "recommendation service" instead of yoochoose" in the change below. Which one is the right one ? :)